autopilot/agent.go

```diff
@@ -197,8 +197,9 @@ type chanCloseUpdate struct {
 // the backing wallet changes.
 func (a *Agent) OnBalanceChange(delta btcutil.Amount) {
 	go func() {
-		a.stateUpdates <- &balanceUpdate{
-			balanceDelta: delta,
+		select {
+		case a.stateUpdates <- &balanceUpdate{balanceDelta: delta}:
+		case <-a.quit:
```

Comment:
👍 

Also add these to the `waitGroup`?

Reply:
Fixed.